kore/src/Kore/Syntax/And.hs

```diff
@@ -73,3 +77,14 @@ instance Unparse child => Unparse (And Sort child) where
             , unparse2 andFirst
             , unparse2 andSecond
             ])
+
+instance Ord variable => Synthetic (And sort) (FreeVariables variable) where
+    synthetic = Foldable.fold
+    {-# INLINE synthetic #-}
+
+instance Synthetic (And Sort) Sort where
+    synthetic And { andSort, andFirst, andSecond } =
+        andSort
+        & seq (matchSort andSort andFirst)
```

Comment:
So, just to make sure: these `matchSort` calls are only used as sanity checks, right?

Reply:
That's right, it just throws an exception if the sorts don't match. Ordinarily that can't happen, but it might when we are constructing patterns in the code.